File: src/main/java/com/challenge/interview/calendar/model/Role.java
package com.challenge.interview.calendar.model;

public enum Role {
    CANDIDATE("candidate"),
    INTERVIEWER("interviewer");


    private String name;
    private Role(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: src/main/kotlin/com/challenge/interview/calendar/model/Role.kt
package com.challenge.interview.calendar.model

import java.util.*
import org.springframework.core.convert.converter.Converter;

enum class Role(val role: String) {
    CANDIDATE("candidate"),
    INTERVIEWER("interviewer");

    override fun toString(): String {
        return this.role;
    }
}

class StringToEnumConverter : Converter<String?, Role?> {
    override fun convert(source: String): Role {
        return Role.valueOf(source.uppercase(Locale.getDefault()))
    }
}